crates/iceberg/src/spec/table_properties.rs

```diff
@@ -175,6 +196,19 @@ impl TryFrom<&HashMap<String, String>> for TableProperties {
                 TableProperties::PROPERTY_WRITE_TARGET_FILE_SIZE_BYTES,
                 TableProperties::PROPERTY_WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT,
             )?,
+            avro_compression_codec: parse_property(
+                props,
+                TableProperties::PROPERTY_AVRO_COMPRESSION_CODEC,
+                TableProperties::PROPERTY_AVRO_COMPRESSION_CODEC_DEFAULT.to_string(),
+            )?,
+            avro_compression_level: {
+                let level = parse_property(
+                    props,
+                    TableProperties::PROPERTY_AVRO_COMPRESSION_LEVEL,
+                    255u8,
+                )?;
+                if level == 255 { None } else { Some(level) }
+            },
```

Comment:
is this not valid here? 
```
            avro_compression_level: parse_property(
                props,
                TableProperties::PROPERTY_AVRO_COMPRESSION_LEVEL,
                TableProperties::PROPERTY_AVRO_COMPRESSION_LEVEL_DEFAULT,
            )?,
```

`avro_compression_level` and `PROPERTY_AVRO_COMPRESSION_LEVEL_DEFAULT` are both `Option<u8>`

Reply:
I might be misunderstanding the comment.  But  I should have cleaned up `pub const PROPERTY_AVRO_COMPRESSION_LEVEL_DEFAULT: Option<u8> = None`;, I don't think it is used anymore.

In a prior revision I added a method (`parse_optional_property`) that could return  `None` value directly but @liurenjie1024 questioned whether this was useful.